pkg/actions/lua/lakefs/catalogexport/hive.lua

```diff
@@ -0,0 +1,68 @@
+local pathlib = require("path")
+local utils = require("lakefs/catalogexport/internal/utils")
+
+-- extract partition prefix from full path
+function extract_partitions_path(partitions, path)
```

Comment:
suggest - assume the path starts with the partition pattern "<col>=<val>/..."
```lua
local function extract_partitions_path(partitions, path)
    if path == "" then
        return ""
    end
    local idx = 1
    for _, partition in ipairs(partitions) do
        local expected = partition .. "="
        local key = string.sub(path, idx, idx + string.len(expected) - 1)
        if key ~= expected then
            return nil
        end
        idx = idx + expected:len()
        local i = string.find(path, "/", idx)
        if i == nil then
            return nil
        end
        idx = i + 1
    end
    return string.sub(path, 1, idx - 1)
end
```

Reply:
if the path include the prefix + partition + filename, like `tables/animals1/type=axolotl/weight=12/name=doe/animals.csv`
we are missing the first `/` check as the current code will do partial match for column name `a` where the path include `aaa=v1/`

Reply:
please ignore the above I see that you handle `/` case